### PR TITLE
Add collector for the sudoers table

### LINKED ISSUE #29629: Add collector for the sudoers table
## Environment

| Wazuh version | Install type | Install method     | Platform       |
|---------------|--------------|--------------------|----------------|
| 4.14.0        | Agent        | Packages / Sources | Linux, macOS|

# Description

Implement support for collecting sudo rule definitions from the operating system using the `extended_sources` submodule in `data_provider`. This data corresponds to the `sudoers` table and allows Wazuh to track sudo rules as they are defined across source files in Linux and macOS systems. The information contributes to understanding privilege escalation configurations.

**Relevant fields:**

| Column        | Type | Description                               |
|---------------|------|-------------------------------------------|
| `source`      | text | Source file containing the given rule     |
| `header`      | text | Symbol for given rule                     |
| `rule_details`| text | Rule definition                           |

# Tasks

- [ ] Implement the data extractor for Linux and macOS.  
- [ ] Implement logic in `extended_sources` to retrieve sudoers data.  
- [ ] Add unit tests to validate field extraction and transformation.  

# Acceptance Criteria

- Results are structured as stateful in JSON.  
- Field names align with ECS where applicable.  
- Tests cover Linux/macOS behavior and edge cases.  
- Supported platforms: Linux, macOS.

# References

- [Internal analysis](https://docs.google.com/spreadsheets/d/1XwgDInDeReqhZ6DoAn4SmqrKvCFmN40NjoUBWQiwt54/edit#gid=0)  
- [Schema documentation](https://osquery.io/schema/5.16.0/#sudoers)  
- [Spec definition](https://github.com/osquery/osquery/blob/master/specs/posix/sudoers.table)  
- [Table grouping reference](https://github.com/wazuh/wazuh-agent/issues/664#issuecomment-2769043629)

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|closes #29629|

## Environment

| Wazuh version | Install type | Install method     | Platform       |
|---------------|--------------|--------------------|----------------|
| 4.14.0        | Agent        | Packages / Sources | Linux, macOS|

# Description

Implement support for collecting sudo rule definitions from the operating system using the `extended_sources` submodule in `data_provider`. This data corresponds to the `sudoers` table and allows Wazuh to track sudo rules as they are defined across source files in Linux and macOS systems. The information contributes to understanding privilege escalation configurations.

**Relevant fields:**

| Column        | Type | Description                               |
|---------------|------|-------------------------------------------|
| `source`      | text | Source file containing the given rule     |
| `header`      | text | Symbol for given rule                     |
| `rule_details`| text | Rule definition                           |

# Tasks

- [ ] Implement the data extractor for Linux and macOS.  
- [ ] Implement logic in `extended_sources` to retrieve sudoers data.  
- [ ] Add unit tests to validate field extraction and transformation.  

# Acceptance Criteria

- Results are structured as stateful in JSON.  
- Field names align with ECS where applicable.  
- Tests cover Linux/macOS behavior and edge cases.  
- Supported platforms: Linux, macOS.

# References

- [Internal analysis](https://docs.google.com/spreadsheets/d/1XwgDInDeReqhZ6DoAn4SmqrKvCFmN40NjoUBWQiwt54/edit#gid=0)  
- [Schema documentation](https://osquery.io/schema/5.16.0/#sudoers)  
- [Spec definition](https://github.com/osquery/osquery/blob/master/specs/posix/sudoers.table)  
- [Table grouping reference](https://github.com/wazuh/wazuh-agent/issues/664#issuecomment-2769043629)

### CODE REVIEW

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+void SudoersProvider::genSudoersFile(const std::string& fileName,
+                                     unsigned int level,
+                                     nlohmann::json& results)
+{
+    // sudoers(5): No more than 128 files are allowed to be nested.
+    static const unsigned int kMaxNest = 128;
+
+    // Local utility functions/lambdas
+    auto trim = [](std::string & s)
```
Comment (Nicogp) [inline src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp:51]: I would consider moving the lambdas to another utils file, to allow their reuse in other cases. Or perhaps the function in `shared_modules/utils/stringHelper.h` can be reused.

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        {
+            return !std::isspace(ch);
+        }));
+        s.erase(std::find_if(s.rbegin(), s.rend(), [](int ch)
+        {
+            return !std::isspace(ch);
+        }).base(), s.end());
+    };
+
+    auto listFilesInDirectory = [](const std::string & path, std::vector<std::string>& out)
```
Comment (Nicogp) [inline src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp:63]: Same. Maybe in the file `shared_modules/utils/fileSystem.h` or `shared_modules/utils/filesystemHelper.h` . Function `enumerateDir ` of `shared_modules/utils/filesystemHelper.h`?.

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+        if (isInclude)
+        {
+            // Relative or full paths
+            if (ruleDetails.at(0) != '/')
+            {
+                ruleDetails = resolvePath(ruleDetails);
+            }
+
+            genSudoersFile(ruleDetails, ++level, results);
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp]: indeed, changed!

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+            isLongLine = true;
+        }
+
+        nlohmann::json entry;
+        entry["header"] = header;
+        entry["source"] = fileName;
+        entry["rule_details"] = ruleDetails;
+        results.push_back(std::move(entry));
+
+        auto resolvePath = [&](const std::string & relativePath)
```
Comment (Nicogp) [inline src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp:185]: Same. Maybe in the file `shared_modules/utils/fileSystem.h` or `shared_modules/utils/filesystemHelper.h`

**Review by cborla — APPROVED** [review]
LGTM
- [x] Code changes reviewed
- [x] Relevant evidence provided
- [x] Tests cover the new functionality
- [x] Meets requirements and/or definition of done
- [x] No unresolved dependencies with other issues

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp (hunk 2 above)
Comment (LucioDonda) [reply src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp]: Done!

File: src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp (hunk 4 above)
Comment (LucioDonda) [reply src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp]: Done!
